tests/fault_tolerance/migration/utils.py

```diff
@@ -158,8 +170,10 @@ def poll_worker(worker: ManagedProcess, result_list: list[bool]):
         return worker2, "Worker 2"
     elif worker1_received and worker2_received:
         pytest.fail("Both workers received the request")
+        raise AssertionError("Unreachable")  # For mypy: pytest.fail() raises
     else:
         pytest.fail("Neither worker received the request")
+        raise AssertionError("Unreachable")  # For mypy: pytest.fail() raises
```

Comment:
I don't think we need the additional `raise` statement, because the `pytest.fail` above will "return" without reaching the `raise` statement below.
```suggestion
    elif worker1_received and worker2_received:
        pytest.fail("Both workers received the request")
    else:
        pytest.fail("Neither worker received the request")
```